envi/registers.py

```diff
@@ -362,7 +362,7 @@ def _xlateToNativeReg(self, index, value):
         if offset:
             value <<= offset
 
-        return value | (curval & finalmask)
+        return (value & mask) | (curval & finalmask)
```

Comment:
Since this is just a fundamental change, is there a test you could add to show that this works?

Reply:
makes sense :+1: